internal/cnpi/plugin/client/contracts.go

```diff
@@ -67,9 +67,12 @@ type ClusterCapabilities interface {
 	// be changed from a value to another
 	ValidateClusterUpdate(
 		ctx context.Context,
-		oldObject client.Object,
-		newObject client.Object,
+		cluster client.Object,
+		mutatedCluster client.Object,
 	) (field.ErrorList, error)
+
+	// SetClusterStatus returns a map of [pluginName]: statuses to be assigned to the cluster
+	SetClusterStatus(ctx context.Context, cluster client.Object) (map[string]string, error)
```

Comment:
This name is really weird since it's not setting the status of a cluster but the status of a plugin, wouldn't be better to have something like `SetPluginStatus` ? what do you think @armru @leonardoce

Reply:
The GRPC was defined while trying to maintain coherence with the other `operator` capabilities.

Given that the plugin is invoking that capability and is setting its status inside the cluster is not really that misleading, but I'm open for eventual changes

Reply:
The status is inside the cluster so you understand is the status regarding to that plugin in that status, I don't see that like the issue, the issue is that "SetClusterStatus" implies that will set the status of the cluster, not the plugin in that cluster, but I see that this is already defined in the cnpg-i so there's nothing to do here

Reply:
Would SetStatusInCluster be better?
We can still modify cnpg-i given that this gprc is not used yet

Reply:
I think that `SetClusterStatus` is a coherent name. If we want to make it more explicit, we could add `Plugin` to the name and use `SetClusterPluginStatus`.

Reply:
After a brief chat with @sxd we agreed that the best name to avoid ambiguity is `SetStatusInCluster` as proposed by @armru.

Reply:
Yup! came to update too and was already updated :D